common/src/main/java/com/microsoft/identity/common/internal/logging/DiagnosticContext.java

```diff
@@ -22,74 +22,7 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.logging;
 
-public final class DiagnosticContext {
-
-    public static final String CORRELATION_ID = "correlation_id";
-    public static final String THREAD_ID = "thread_id";
-    public static final String THREAD_NAME = "thread_name";
-
-    private DiagnosticContext() {
-    }
-
-    private static final ThreadLocal<IRequestContext> REQUEST_CONTEXT_THREAD_LOCAL =
-            new ThreadLocal<IRequestContext>() {
-                @Override // This is the default value for the RequestContext if it's unset
-                protected RequestContext initialValue() {
-                    final RequestContext defaultRequestContext = new RequestContext();
-                    defaultRequestContext.put(CORRELATION_ID, "UNSET");
-                    return defaultRequestContext;
-                }
-            };
-
-    /**
-     * Set the request context.
-     *
-     * @param requestContext IRequestContext
-     */
-    public static void setRequestContext(final IRequestContext requestContext) {
-        if (null == requestContext) {
-            clear();
-            return;
-        }
-
-        requestContext.put(THREAD_ID, String.valueOf(Thread.currentThread().getId()));
-        requestContext.put(THREAD_NAME, Thread.currentThread().getName());
-        REQUEST_CONTEXT_THREAD_LOCAL.set(requestContext);
-    }
-
-    /**
-     * Get the request context.
-     *
-     * @return IRequestContext
-     */
-    public static IRequestContext getRequestContext() {
-        if (!hasThreadMetadata()) {
-            setThreadMetadata();
-        }
-
-        return REQUEST_CONTEXT_THREAD_LOCAL.get();
-    }
-
-    private static void setThreadMetadata() {
-        REQUEST_CONTEXT_THREAD_LOCAL.get().put(
-                THREAD_ID,
-                String.valueOf(Thread.currentThread().getId())
-        );
-        REQUEST_CONTEXT_THREAD_LOCAL.get().put(
-                THREAD_NAME,
-                Thread.currentThread().getName()
-        );
-    }
-
-    private static boolean hasThreadMetadata() {
-        return REQUEST_CONTEXT_THREAD_LOCAL.get().containsKey(THREAD_ID);
-    }
-
-    /**
-     * Clear the local request context thread.
-     */
-    public static void clear() {
-        REQUEST_CONTEXT_THREAD_LOCAL.remove();
-    }
-
+@Deprecated
+public final class DiagnosticContext extends com.microsoft.identity.common.logging.DiagnosticContext {
+    // Class exists for backcompat - originally internal only
```

Comment:
You probably want implementations of all the public methods that 1-shot errors into the logs